hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieHFileDataBlock.java

```diff
@@ -162,6 +167,18 @@ protected void createRecordsFromContentBytes() throws IOException {
     return records;
   }
 
+  /**
+   * Serialize the record to byte buffer.
+   *
+   * @param record         - Record to serialize
+   * @param schemaKeyField - Key field in the schema
+   * @return Serialized byte buffer for the record
+   */
+  protected ByteBuffer serializeRecord(final IndexedRecord record, final Option<Field> schemaKeyField) {
+    ValidationUtils.checkArgument(record.getSchema() != null, "Unknown schema for the record!");
+    return ByteBuffer.wrap(HoodieAvroUtils.indexedRecordToBytes(record));
```

Comment:
What is the benefit of using ByteBuffer.wrap here? We do not perform any operations on the returned ByteBuffer but instead call .array() on it to retrive the internal byte[]

Reply:
Since this serializeRecord() method is intended for overloading by the derived classes, want the interface to be much more expressive than the primitive array type. In the current implementation, ByteBuffer.wrap() is only a "view" of the backing single copy of the record byte array and has the same performance as that of byte[] . Additionally, the caller can work on the returned ByteBuffer to peek/parse/duplicate the content as needed. Right, we aren't using any of this in the current implementation. I am ok to change this to byte[].